test/gtest/uct/test_peer_failure.h

```diff
@@ -0,0 +1,226 @@
+
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+#include "uct_test.h"
+#include <common/test.h>
+extern "C" {
+#include <uct/api/uct.h>
+#include <uct/sm/mm/base/mm_ep.h>
+#include <uct/sm/scopy/cma/cma_ep.h>
+}
+
+#include <vector>
+
+
+class test_uct_peer_failure : public uct_test {
+private:
+    struct am_handler_setter
+    {
+        am_handler_setter(test_uct_peer_failure *test) : m_test(test) {}
+
+        void operator() (test_uct_peer_failure::entity *e) {
```

Comment:
can we move body of all functions to CC file?

Reply:
moved functions (except simple getters/setters) into CC file